src/lib/components/Error/WidgetPropValidator.tsx

```diff
@@ -0,0 +1,36 @@
+import { WidgetProps } from 'lib/components/Widget'
+import { IntegrationError } from 'lib/errors'
+import { PropsWithChildren, useEffect, useMemo, useState } from 'react'
+
+export default function WidgetPropValidator(props: PropsWithChildren<WidgetProps>) {
+  const { jsonRpcEndpoint, provider } = props
+
+  const [providerChecked, setProviderChecked] = useState(false)
```

Comment:
- Why can't this be checked every time?
- Instead of introducing state, you can set the effect to run only on mount by not passing a deps array (or passing an empty array). If the linter stops you, disable the check and leave a comment explaining the behavior.

This applies to all checks. I don't think we need ot introduce local state to only check them once. What if the props change, and future values are invalid?

Reply:
These checks are run every time. The state does not prevent re-running the hook. It exists to keep the child dom tree from re-rendering with a potentially invalid config.